pandas/core/indexes/extension.py

```diff
@@ -187,6 +187,9 @@ class NDArrayBackedExtensionIndex(ExtensionIndex):
     def _get_engine_target(self) -> np.ndarray:
         return self._data._ndarray
 
+    def _get_join_target(self) -> np.ndarray:
```

Comment:
could return `self._get_engine_target()` here and not have to override in datetimelike

Reply:
Good idea